src/casl/casl-ability.factory.ts

```diff
@@ -783,96 +784,175 @@ export class CaslAbilityFactory {
+      })
+      can(AuthOp.JobCreateConfiguration,JobClass, {
+        'configuration.create.auth' : '#datasetPublic',
+        'datasetsValidation': true
+      })
+    } else {
       /**
-       * authenticated users belonging to any of the group listed in CREATE_JOBS_GROUPS
+       * authenticated users
        */
 
-      // -------------------------------------
-      // jobs
-      // -------------------------------------
-      // endpoint authorization
-      can(AuthOp.JobsRead, JobClass);
-      can(AuthOp.JobsCreate, JobClass);
-      can(AuthOp.JobsUpdate, JobClass);
-      // -------------------------------------
-      // data instance authorization
-      can(AuthOp.JobsCreateAny, JobClass, {
-        ownerGroup: { $in: user.currentGroups },
-      });
-      can(AuthOp.JobsReadAccess, JobClass, {
-        ownerGroup: { $in: user.currentGroups },
-      });
-      can(AuthOp.JobsUpdateAny, JobClass, {
-        ownerGroup: { $in: user.currentGroups },
-      });
-    } else if (
-      user.currentGroups.some((g) =>
-        configuration().updateJobGroups.includes(g),
-      )
-    ) {
-      /**
-       * authenticated users belonging to any of the group listed in UPDATE_JOBS_GROUPS
-       */
+      // check if this user is part of the admin group
+      if (
+        user.currentGroups.some((g) => configuration().adminGroups.includes(g))
+      ) {
+        /**
+         * authenticated users belonging to any of the group listed in ADMIN_GROUPS
+         */
 
-      // -------------------------------------
-      // jobs
-      // -------------------------------------
-      // endpoint authorization
-      cannot(AuthOp.JobsRead, JobClass);
-      cannot(AuthOp.JobsCreate, JobClass);
-      can(AuthOp.JobsUpdate, JobClass);
-      // -------------------------------------
-      // data instance authorization
-      can(AuthOp.JobsUpdateAny, JobClass, {
-        ownerGroup: { $in: user.currentGroups },
-      });
-    } else if (user) {
-      /**
-       * authenticated users
-       */
+        // -------------------------------------
+        // endpoint authorization
+        can(AuthOp.JobRead, JobClass);
+        can(AuthOp.JobCreate, JobClass);
+        can(AuthOp.JobStatusUpdate, JobClass);
+      
+        // -------------------------------------
+        // data instance authorization
+        can(AuthOp.JobReadAny, JobClass);
+        can(AuthOp.JobCreateAny, JobClass);
+        can(AuthOp.JobStatusUpdateAny, JobClass);
 
-      // -------------------------------------
-      // jobs
-      // -------------------------------------
-      // endpoint authorization
-      can(AuthOp.JobsRead, JobClass);
-      cannot(AuthOp.JobsCreate, JobClass);
-      cannot(AuthOp.JobsUpdate, JobClass);
-      // -------------------------------------
-      // data instance authorization
-      can(AuthOp.JobsReadAccess, JobClass, {
-        ownerGroup: { $in: user.currentGroups },
-      });
+      } else {
```

Comment:
How can we test all the following code?

Reply:
We need to add a functional account that is only part of Job Group.
We can take this off-line